src/silx/io/url.py

```diff
@@ -277,7 +283,10 @@ def __parse_from_path(self, path: str):
             file_path = self.__path[0:pos] + url.path
         else:
             scheme = url.scheme if url.scheme != "" else None
-            file_path = url.path
+            if url.netloc != "":
+                file_path = f"{url.netloc}{url.path}"
+            else:
+                file_path = url.path
```

Comment:
This can be simplified to:
```suggestion
            file_path = f"{url.netloc}{url.path}"
```